binary_port/src/binary_message.rs

```diff
@@ -0,0 +1,217 @@
+#[cfg(test)]
+use casper_types::testing::TestRng;
+#[cfg(test)]
+use rand::Rng;
+
+use bytes::Buf;
+use tokio_util::codec::{self};
+
+use crate::error::Error;
+
+type LengthEncoding = u32;
+const LENGTH_ENCODING_SIZE_BYTES: usize = std::mem::size_of::<LengthEncoding>();
+
+#[derive(Clone, PartialEq, Debug)]
+pub struct BinaryMessage(Vec<u8>);
+
+impl BinaryMessage {
+    pub fn new(payload: Vec<u8>) -> Self {
+        BinaryMessage(payload)
```

Comment:
nitpick but
```suggestion
        Self(payload)
```

Reply:
Updated in e2c991e79b4eaed6fb7d9d3d1ab8b8298e1671c1